zerver/views/realm.py

```diff
@@ -158,6 +167,19 @@ def update_realm(
     if invite_to_realm_policy is not None and not user_profile.is_realm_owner:
         raise OrganizationOwnerRequired()
 
+    if is_demo_organization is not None and not is_demo_organization:
+        if not user_profile.is_realm_owner:
+            raise OrganizationOwnerRequired()
+
+        if realm.demo_organization_scheduled_deletion_date is None:
+            raise JsonableError(_("Organization to be converted must be a demo organization."))
+
+        if string_id is None:
+            raise JsonableError(_("Organization subdomain is required."))
+
+        if string_id is not None and realm.demo_organization_scheduled_deletion_date is not None:
+            do_convert_demo_realm_to_regular_realm(realm, string_id)
```

Comment:
This needs to call `check_subdomain_available` -- otherwise you can override another organization's subdomain, or a restricted one.

From an API perspective, I think we could just have the a `string_id` parameter -- it's implied that if you're a demo organization and want to change your subdomain, that you want to convert to a real organization, and lets us simplify the API.  

(Also, if we start allowing folks to change their subdomain, it'd be able to use the same interface).

Reply:
Ooh that's a great idea!